### PR TITLE
Implement Redis 7.0.0 [P]EXPIRE[AT] options

### PR DESCRIPTION
**Pull request by michael-grunder**
See #2068

### CODE REVIEW

**Comment by michael-grunder** [comment]
@yatsukhnenko I wasn't really sure the best way to handle the new option.

This way is flexible, but I wonder if we should instead take an options array for future-proofing (e.g. if Redis adds more exotic options to this command).

**Comment by yatsukhnenko** [comment]
@michael-grunder currently in `set` and `zadd` commands we use array to handle NX/EX/etc options so maybe it worth to implement it here but I'm totally fine with current implementation.

File: redis_commands.c [hunk 1]
```diff
@@ -5724,6 +5724,32 @@ int redis_xtrim_cmd(INTERNAL_FUNCTION_PARAMETERS, RedisSock *redis_sock,
...
+{
+    zend_string *key = NULL, *mode = NULL;
+    smart_string cmdstr = {0};
+    zend_long timeout = 0;
+
+    ZEND_PARSE_PARAMETERS_START(2, 3)
+        Z_PARAM_STR(key)
+        Z_PARAM_LONG(timeout)
+        Z_PARAM_OPTIONAL
+        Z_PARAM_STR(mode)
```
Comment (yatsukhnenko) [inline redis_commands.c:5740]: Consider to add check for `mode` values

**Review by yatsukhnenko — APPROVED** [review]

**Comment by michael-grunder** [comment]
@yatsukhnenko I added a check on the mode argument and updated our stubs/docs with actual documentation.

I think maybe we should break this command in PhpRedis 6.0.0 to instead return an integer (`0` when an expiration was not set, `1` when an expiration was set, and `false` on failure).

**Pull request merged**